File: alternating_elements.py
#swap 2 elements in list
def swap(list,pos1,pos2):
    list[pos1],list[pos2]=list[pos2],list[pos1]
    return list
    

def reorder(numbers):
    for i in range(1,len(numbers)-1):
        if((not(i%2) and numbers[i-1] < numbers[i]) or ((i%2) and numbers[i-1] > numbers[i])):
            swap(numbers,i,i-1)


lst=[6,1,8,12,88,12,3,13,10]   
reorder(lst)
print(lst)

File: alternating_elements.cpp
#include <algorithm>
#include <vector>
//#include <iostream>
//using std::cout;
using std::vector;

std::vector<int> reorder(std::vector<int>& numbers)
{  
    std::sort(numbers.begin(),numbers.end());
    for(size_t i=1;i<numbers.size()-1;i+=2)
    {
        std::swap(numbers[i],numbers[i+1]);
    }
    return numbers;
}